loadtest/loadtest.go

```diff
@@ -0,0 +1,334 @@
+		tcClients:    config.TCClients,
+		tcHelmClient: config.TCHelmClient,
+
+		clusterID:            config.ClusterID,
+		commonDomain:         config.CommonDomain,
+		stormForgerAuthToken: config.StormForgerAuthToken,
+	}
+
+	return s, nil
+}
+
+func (l *LoadTest) Test(ctx context.Context) error {
+	var err error
+
+	var loadTestEndpoint string
+	{
+		loadTestEndpoint = fmt.Sprintf("loadtest-app.%s.%s", l.clusterID, l.commonDomain)
+
+		l.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("loadtest-app endpoint is %#q", loadTestEndpoint))
+	}
+
+	{
+		l.logger.LogCtx(ctx, "level", "debug", "message", "waiting for tenant cluster kubernetes API to be up")
+
+		err = l.waitForAPIUp(ctx)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		l.logger.LogCtx(ctx, "level", "debug", "message", "waited for tenant cluster kubernetes API to be up")
+	}
+
+	{
+		l.logger.LogCtx(ctx, "level", "debug", "message", "installing loadtest app")
+
+		err = l.installTestApp(ctx, loadTestEndpoint)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		l.logger.LogCtx(ctx, "level", "debug", "message", "installed loadtest app")
+	}
+
+	{
+		l.logger.LogCtx(ctx, "level", "debug", "message", "waiting for loadtest app to be ready")
+
+		err = l.waitForLoadTestApp(ctx)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		l.logger.LogCtx(ctx, "level", "debug", "message", "loadtest app is ready")
+	}
+
+	{
+		l.logger.LogCtx(ctx, "level", "debug", "message", "enabling HPA for Nginx Ingress Controller")
+
+		err = l.enableIngressControllerHPA(ctx)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		l.logger.LogCtx(ctx, "level", "debug", "message", "enabled HPA for Nginx Ingress Controller")
+	}
+
+	return nil
+}
+
+// enableIngressControllerHPA enables HPA via the user configmap and updates
+// the chartconfig CR so chart-operator reconciles the config change.
+func (l *LoadTest) enableIngressControllerHPA(ctx context.Context) error {
+	var err error
+
+	l.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waiting for %#q configmap to be created", UserConfigMapName))
+
+	o := func() error {
+		lo := metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("app=%s,cluster-operator.giantswarm.io/configmap-type=user", UserConfigMapName),
+		}
+		l, err := l.tcClients.K8sClient().CoreV1().ConfigMaps(metav1.NamespaceDefault).List(lo)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if len(l.Items) != 1 {
+			return microerror.Maskf(waitError, "want %d configmaps found %d", 1, len(l.Items))
+		}
+
+		return nil
+	}
+
+	b := backoff.NewConstant(10*time.Minute, 15*time.Second)
+	n := func(err error, delay time.Duration) {
+		l.logger.LogCtx(ctx, "level", "debug", "message", err.Error())
+	}
+
+	err = backoff.RetryNotify(o, b, n)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	l.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waited for %#q configmap to be created", UserConfigMapName))
```

Comment:
We could put the code up to here into a scope block. Would make it easier to scan the different parts of the function, at least for me.

Reply:
Yes that's nicer, thx. I also added some missing logging.